### ISSUE TITLE
error in implementing dim_aware_loss

### ISSUE THREAD

**Issue by zhaokai5** [xinzhuma/monodle#30]
Hi,thanks for your great work. when we use dim_aware_loss to train dim we found that there is some bug in dim_aware_loss.

https://github.com/xinzhuma/monodle/blob/e426aa65fdc7ceedcaab0d637acf3d3425d0736c/lib/losses/dim_aware_loss.py#L8

loss /= dimension may change the gradient direction of one of the parameters (h, w, l). For example, when h < 0, the gradient is of |h - h*| is negative, so that the graident changes to positive when divided by h and this will cause the loss to increase when using gradient descent to update parameters. The gradient direction of all parameters should not be changed after dividing by dimension. We recommand that this line be changed to loss/=torch.abs(dimension).
